### PR TITLE
Introduce contiv as a CNI provider

### LINKED ISSUE #671: Add Contiv as a CNI provider
Introduce support for Contiv as a CNI provider.

- [x] Figure out if we can use etcd v3
- [x] Figure out if we can use secure etcd endpoints (how to pass certificates to wherever they are needed): 
Doesn't look like contiv currently supports an etcd cluster over HTTPS,
 as it uses the default HTTP transport in the etcd client,
 and there is not way to override it.
 (https://github.com/contiv/netplugin/blob/master/state/etcdstatedriver.go#L62).
- [x] Figure out if contiv supports multiple etcd nodes
Contiv currently doesn't support multiple etcd nodes in the backend. It only sets one URL when building the etcd client:
 (https://github.com/contiv/netplugin/blob/master/state/etcdstatedriver.go#L63).
- [ ] Translate installation script into an ansible play
- [x] Figure out what to do about taints - the taint is used for a replicaset that should only run on master nodes. We can probably rework this to use a daemonset.
- [x] "Own" the resources that get deployed
- [ ] How to ship the contiv binary (`netctl`) - probably use a container similar to what we'll do with CNI

### PR DESCRIPTION
**Pull request by alexbrand**
Fixes #671 

To do:
- [x] Move cni installation to container
- [x] Figure out how to distribute netctl
- [x] Add integration test coverage
- [x] Figure out issue with kuberang
- [x] Move netmaster to daemonset that gets deployed on master nodes

We'll do this in follow up PRs:
- [ ] Review contiv options we want to expose in plan file
- [ ] Figure out upgrades with contiv

### CODE REVIEW

File: ansible/_contiv.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+  - hosts: master:worker:ingress:storage
+    any_errors_fatal: true
+    name: "{{ play_name | default('Start Contiv Network Components') }}"
+    serial: "{{ serial_count | default('100%') }}"
+    remote_user: root
+    become_method: sudo
```
Comment (dkoshkin) [inline ansible/_contiv.yaml]: We want to use become: yes after #668

File: ansible/roles/contiv/templates/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      "cniVersion": "0.1.0",
+      "name": "contiv-net",
+      "type": "contivk8s"
+    }
+  config: |-
+    {
+       "K8S_API_SERVER": "https://{{ kubernetes_load_balanced_fqdn }}:6443",
+       "K8S_CA": "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+       "K8S_KEY": "",
+       "K8S_CERT": "",
```
Comment (dkoshkin) [inline ansible/roles/contiv/templates/configmap.yaml]: Do we not need `K8S_KEY` and `K8S_CERT`?

Comment (alexbrand) [inline ansible/roles/contiv/templates/configmap.yaml]: It uses the default locations for the key and cert.

**Comment by dkoshkin** [comment]
Should we pre-pull docker images similar to how we do it for Calico? That was originally added because we were seeing timeout of the pods starting on slow networks.

**Comment by alexbrand** [comment]
Definitely. Thanks @dkoshkin!

File: ansible/roles/contiv/tasks/main.yaml [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      command: docker run --net host --rm --entrypoint /contiv/bin/netctl {{ contiv_netplugin_img }} --netmaster http://localhost:9999 net ls
+      register: contiv_networks
+      run_once: true
+
+    - name: create infra network if missing
+      command: docker run --net host --rm --entrypoint /contiv/bin/netctl {{ contiv_netplugin_img }} --netmaster http://localhost:9999 net create -n infra -s 132.1.1.0/24 -g 132.1.1.1 contivh1 # using defaults from installation script
+      when: "'contivh1' not in contiv_networks.stdout"
+      run_once: true
+
+    - name: restart netplugin to propagate forwarding mode change
```
Comment (dkoshkin) [inline ansible/roles/contiv/tasks/main.yaml]: This seems crazy, but if thats what it takes :(

File: ansible/roles/kubelet/templates/kubelet.service [hunk 4]
```diff
@@ -21,6 +21,9 @@ ExecStart=/usr/bin/kubelet \
   --require-kubeconfig=true \
   --kubeconfig={{ kubernetes_kubeconfig.kubelet }} \
   --node-ip={{ internal_ipv4 }} \
+{% if inventory_hostname in groups['master'] %}
+  --node-labels=node-role.kubernetes.io/master= \
```
Comment (dkoshkin) [inline ansible/roles/kubelet/templates/kubelet.service]: Is this a common label?

File: pkg/install/validate_test.go [hunk 5]
```diff
@@ -1005,17 +1005,12 @@ func TestCNIAddOn(t *testing.T) {
 			},
 			valid: true,
 		},
-		// {
```
Comment (dkoshkin) [inline pkg/install/validate_test.go]: Can remove.

File: ansible/roles/contiv/tasks/main.yaml (hunk 3 above)
Comment (alexbrand) [inline ansible/roles/contiv/tasks/main.yaml]: Yup, that's what the official installation script does as well.

File: ansible/roles/kubelet/templates/kubelet.service (hunk 4 above)
Comment (alexbrand) [inline ansible/roles/kubelet/templates/kubelet.service]: There is still some discussion around this label, but both kubeadm and kargo are using it. Discussion here: https://groups.google.com/forum/#!topic/kubernetes-sig-cluster-lifecycle/_Ys5ABKgcbM

**Comment by dkoshkin** [comment]
LGTM!

**Pull request merged**